app/forms/registration_form.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class RegistrationForm < ApplicationForm
+  def save(create_params)
+    return false unless validate_client(create_params[:client_id])
+
+    user = User.new(create_params.except(:client_id))
+
+    if user.invalid?
+      user_errors = user.errors
+      errors.add(:user, user_errors.full_messages.last) unless user_errors.empty?
+
+      return false
+    end
+
+    user.save
+  end
+
+  private
+
+  def validate_client(client_id)
+    unless Doorkeeper::Application.find_by(uid: client_id)
+      errors.add(:client_id, I18n.t('doorkeeper.errors.messages.invalid_client'))
+
+      return
+    end
+
+    true
+  end
```

Comment:
this method returns `true` or `nil`. 
In general, I recommend to return `true/false` or `object_instance/nil`, but not returning a nullable boolean — even if "it works" ^^

Besides, some minor recommendations:
- we can have a more Rails flavor `client_valid?` naming
- we can leverage an `early return` to avoid using `unless` and make the code feel lighter 🪶 
- As we do not use the returned value of `find_by`, it will be more performant to use the [exists?](https://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-exists-3F) method :) 

```suggestion
  def client_valid?(client_id)
    return true if Doorkeeper::Application.exists?(uid: client_id)
    
    errors.add(:client_id, I18n.t('doorkeeper.errors.messages.invalid_client'))
    
    false
  end
```